EIPS/blockhash_refactoring.md

```diff
@@ -41,6 +41,39 @@ If `block.number >= METROPOLIS_FORK_BLKNUM + 256`, then the BLOCKHASH opcode ins
 
 Also, for blocks where `block.number >= METROPOLIS_FORK_BLKNUM`, the gas cost is increased from 20 to 800 to reflect the higher costs of processing the algorithm in the contract code.
 
+
+### Contract model
+
+The contract, when called by SYSTEM account, records provided block hashes
+selectively on different levels. The levels differs by periods (frequency)
+the block hashes are stored.
+
+Let `n` be the current block number.
+Let `p = n - 1` be the previous block number.
+Let `h` be the block hash of the previous `p` block.
+Let `B` be the base -- the number of records kept on every level.
+
+The levels are numerated by `k`, starting from 0. A level block hash period
+is `B<sup>k</sup>`.
```

Comment:
I don't understand the phrase "A level block hash period is B^k". Can you clarify?